### ISSUE TITLE
proxiable-mixin triggers didInsertElement property changed deprecation/performance warning.

### ISSUE THREAD

**Issue by DanChadwick** [miguelcobain/ember-paper#237]
Using Ember 2.1.0, I receive:
>DEPRECATION: A property of &lt;in-control@view:-outlet::ember387&gt; was modified inside the didInsertElement hook. You should never change properties on components, services or models during didInsertElement because it causes significant performance degradation. [deprecation id: ember-views.dispatching-modify-property]

This seems to be caused by this in proxiable-mixin.js:
```javascript
  didInsertElement() {
    this._super(...arguments);

    var proxy = this.nearestOfType(ProxyMixin);
    if (proxy) {
      proxy.register(this); // <--------------- culprit
    }
  },
```
I'm rather new to Ember and ember-paper, so I don't have a suggestion for how to address this. Maybe there is there a way to defer this until after didInsertElement without compromising proper functioning?

**Comment by kgautreaux** [comment]
You may be correct, but this still open issue (emberjs/ember.js/issues/11505) from June of 2015 indicates that the warning links to the wrong view and always blames the `didInsertElement` hook when it could be any code that changes component state during `render` of a component.

In Kris Selden's example he uses the `didRender` hook but the deprecation warning is still the same.  This is on the deprecation documentation todo list (ember/ember.js/issues/12290 -> emberjs/website/issues/2256) so if you learn a great deal about it you could perhaps land a PR for the ember website deprecation guide or even ember itself.  Hope this saves you some spelunking.

**Comment by canufeel** [comment]
@DanChadwick #199 fixes this.

**Comment by DanChadwick** [comment]
This is fixed in wip/v1.0.0 branch.
